build breaks with SIMD in mask4

It seems commit 861af905239f4b114e7b6519f97cdcbf6102febf breaks the builds when using SIMD.
```
In file included from /home/cdc100/deploy/oiio/src/include/OpenImageIO/fmath.h:64:0,
                 from /home/cdc100/deploy/oiio/src/include/OpenImageIO/hash.h:50,
                 from /home/cdc100/deploy/oiio/src/include/OpenImageIO/strutil.h:54,
                 from /home/cdc100/deploy/oiio/src/libutil/argparse.cpp:43:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h: In static member function ‘static const OpenImageIO::v1_7::simd::mask4 OpenImageIO::v1_7::simd::mask4::True()’:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h:284:65: error: cannot convert ‘__m128i {aka __vector(2) long long int}’ to ‘__m128 {aka __vector(4) float}’ for argument ‘1’ to ‘__m128i _mm_castps_si128(__m128)’
         __m128i anyval = _mm_castps_si128 (_mm_undefined_si128());
                                                                 ^
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h: In static member function ‘static const OpenImageIO::v1_7::simd::int4 OpenImageIO::v1_7::simd::int4::NegOne()’:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h:657:65: error: cannot convert ‘__m128i {aka __vector(2) long long int}’ to ‘__m128 {aka __vector(4) float}’ for argument ‘1’ to ‘__m128i _mm_castps_si128(__m128)’
         __m128i anyval = _mm_castps_si128 (_mm_undefined_si128());  
```
Disabling SIMD via **USE_SIMD=0** results in another error
```
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h: In member function ‘void OpenImageIO::v1_7::simd::float4::load(const half*)’:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h:1681:9: error: ‘m_vec’ was not declared in this scope
         m_vec = _mm_cvtph_ps (a);
         ^
In file included from /home/cdc100/deploy/oiio/src/include/OpenImageIO/fmath.h:64:0,
                 from /home/cdc100/deploy/oiio/src/include/OpenImageIO/hash.h:50,
                 from /home/cdc100/deploy/oiio/src/include/OpenImageIO/strutil.h:54,
                 from /home/cdc100/deploy/oiio/src/include/OpenImageIO/ustring.h:137,
                 from /home/cdc100/deploy/oiio/src/libutil/typedesc.cpp:39:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h: In member function ‘void OpenImageIO::v1_7::simd::float4::store(half*) const’:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h:1784:35: error: ‘m_vec’ was not declared in this scope
         __m128i h = _mm_cvtps_ph (m_vec, (_MM_FROUND_TO_NEAREST_INT |_MM_FROUND_NO_EXC));   
```
The references i can find point to a similar issue with ticket #952

--- a/.travis.yml
+++ b/.travis.yml
@@ -114,6 +114,13 @@ matrix:
       - os: osx
         compiler: clang
         env: WHICHGCC=5 BUILD_FLAGS="USE_CPP11=1"
+    include:
+      - os: linux
+        compiler: gcc
+        env: WHICHGCC=5 BUILD_FLAGS="USE_CPP11=1 USE_SIMD=avx,f16c"
+      - os: linux
+        compiler: gcc
+        env: WHICHGCC=4.8 BUILD_FLAGS="USE_CPP11=1 USE_SIMD=0"
 
 notifications:
     email:

--- a/src/include/OpenImageIO/simd.h
+++ b/src/include/OpenImageIO/simd.h
@@ -280,8 +280,8 @@ public:
 #if defined(OIIO_SIMD_SSE)
         // Fastest way to fill an __m128 with all 1 bits is to cmpeq_epi8
         // any value to itself.
-# if defined(OIIO_SIMD_AVX)
-        __m128i anyval = _mm_castps_si128 (_mm_undefined_si128());
+# if defined(OIIO_SIMD_AVX) && (OIIO_GNUC_VERSION > 50000)
+        __m128i anyval = _mm_undefined_si128();
 # else
         __m128i anyval = _mm_castps_si128 (_mm_setzero_ps());
 # endif
@@ -429,7 +429,7 @@ public:
         // Fastest way to bit-complement in SSE is to xor with 0xffffffff.
         // Fastest way to fill an __m128 with all 1 bits is to cmpeq_epi8
         // any value to itself.
-# if defined(OIIO_SIMD_AVX)
+# if defined(OIIO_SIMD_AVX) && (OIIO_GNUC_VERSION > 50000)
         __m128i anyval = _mm_undefined_si128(); // AVX only, sigh
 # else
         __m128i anyval = _mm_castps_si128 (m_vec);
@@ -653,8 +653,8 @@ public:
 #if defined(OIIO_SIMD_SSE)
         // Fastest way to fill an __m128 with all 1 bits is to cmpeq_epi8
         // any value to itself.
-# if defined(OIIO_SIMD_AVX)
-        __m128i anyval = _mm_castps_si128 (_mm_undefined_si128());
+# if defined(OIIO_SIMD_AVX) && (OIIO_GNUC_VERSION > 50000)
+        __m128i anyval = _mm_undefined_si128();
 # else
         __m128i anyval = _mm_castps_si128 (_mm_setzero_ps());
 # endif
@@ -1060,7 +1060,7 @@ public:
         // Fastest way to bit-complement in SSE is to xor with 0xffffffff.
         // Fastest way to fill an __m128 with all 1 bits is to cmpeq_epi8
         // any value to itself.
-# if defined(OIIO_SIMD_AVX)
+# if defined(OIIO_SIMD_AVX) && (OIIO_GNUC_VERSION > 50000)
         __m128i anyval = _mm_undefined_si128(); // AVX only, sigh
         __m128i all_one_bits = _mm_cmpeq_epi8 (anyval, anyval);
 # else
@@ -1676,7 +1676,8 @@ public:
 #ifdef _HALF_H_
     /// Load from an array of 4 half values, convert to float
     OIIO_FORCEINLINE void load (const half *values) {
-#ifdef __F16C__ /* Enabled 16 bit float instructions! */
+#if defined(__F16C__) && defined(OIIO_SIMD_SSE)
+        /* Enabled 16 bit float instructions! */
         __m128i a = _mm_castpd_si128 (_mm_load_sd ((const double *)values));
         m_vec = _mm_cvtph_ps (a);
 #elif defined(OIIO_SIMD_SSE) && OIIO_SIMD_SSE >= 2
@@ -1780,7 +1781,7 @@ public:
 
 #ifdef _HALF_H_
     OIIO_FORCEINLINE void store (half *values) const {
-#if __F16C__
+#if defined(__F16C__) && defined(OIIO_SIMD_SSE)
         __m128i h = _mm_cvtps_ph (m_vec, (_MM_FROUND_TO_NEAREST_INT |_MM_FROUND_NO_EXC));
         _mm_store_sd ((double *)values, _mm_castsi128_pd(h));
 #else

